sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -444,6 +472,61 @@ object GpuToTimestamp extends Arm {
     "dd/MM"
   )
 
+  // We are compatible with Spark for these formats when the timeParserPolicy is LEGACY. It
+  // is possible that other formats may be supported but these are the only ones that we have
+  // tests for.
+  val LEGACY_COMPATIBLE_FORMATS = Seq(
```

Comment:
since we are doing lookups in this `Seq` can we make it a `Set[LegacyParseFormat]` or even better a `Map[String, LegacyParseFormat]`

Reply:
Thanks. Fixed.